qiskit_experiments/data_processing/processor_library.py

```diff
@@ -12,36 +12,101 @@
-        normalize: Add a data normalization node to the Kerneled data processor.
+        experiment_data: The experiment data that holds all the data and metadata needed
+             to determine the data processor to use to process the data for analysis.
+        analysis_options: The analysis options with which to analyze the data. The options that
+             are relevant for the configuration of a data processor are:
+             - normalization (bool): A boolean to specify if the data should be normalized to
+               the interval [0, 1]. The default is True. This option is only relevant if
+               kerneled data is used.
+             - outcome (string): The measurement outcome that will be passed to a Probability node.
+             - restless_threshold (float): If (repetition delay / T1) is below a given threshold for
+               all physical qubit T1 values the experiment is recognized as a restless experiment and
+               will be analyzed accordingly.
+        index: The index of the job for which to get a data processor.
 
     Returns:
-        An instance of DataProcessor capable of dealing with the given options.
+        An instance of DataProcessor capable of processing the data for the corresponding job.
+
+    Notes:
+        The following relevant arguments are extracted from the experiment_data metadata run options:
+            - meas_level (MeasLevel): The measurement level of the data to process.
+            - meas_return (MeasReturnType): The measurement return (single or avg) of the data to process.
+            - init_qubits (bool): If False, the qubits are not reset to the ground state after a measurement.
+            - physical qubits: The physical qubits used in the experiment.
+            - memory (bool): If True, single-shot measurement bitstrings are returned.
+            - rep_delay (float): The delay between a measurement and the subsequent circuit.
+
+        In addition, the following argument is extracted from the experiment_data:
+            - t1_values (List[float]): The T1 values of the physical qubits at the time of the experiment.
 
     Raises:
         DataProcessorError: if the measurement level is not supported.
+        DataProcessorError: if no single-shot memory is present but the run options suggest that a
+                            restless experiment was run.
     """
+
+    run_options = experiment_data.metadata["job_metadata"][index].get("run_options", {})
+
+    meas_level = run_options.get("meas_level", MeasLevel.CLASSIFIED)
+    meas_return = run_options.get("meas_return", MeasReturnType.AVERAGE)
+    normalize = analysis_options.get("normalization", True)
+
+    physical_qubits = experiment_data.metadata["physical_qubits"]
+    num_qubits = len(physical_qubits)
+    t1_values = [
+        experiment_data.backend.properties().qubit_property(physical_qubit)["T1"][0]
+        for physical_qubit in physical_qubits
+    ]
```

Comment:
Can you move this right after the if statement 
```
if rep_delay and not init_qubits:
```
so that we do not access the properties if `init_qubits` is `True`.